helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1025,6 +1026,25 @@ public void dropResource(String clusterName, String resourceName) {
     accessor.removeProperty(keyBuilder.resourceConfig(resourceName));
   }
 
+  @Override
+  public void addCloudConfig(String clusterName, CloudConfig cloudConfig) {
+    logger.info("Add CloudConfig to cluster {}, CloudConfig is {}.", clusterName,
+        cloudConfig.toString());
+
+    if (!ZKUtil.isClusterSetup(clusterName, _zkClient)) {
+      throw new HelixException("cluster " + clusterName + " is not setup yet");
+    }
+
+    CloudConfig.Builder builder = new CloudConfig.Builder(cloudConfig);
```

Comment:
Why do we need to copy build a new object for this logic? If it is for validation, please make that method callable here. A builder should not be used for validation.

Reply:
Addressed.

Reply:
This is not we discussed, right? We just need to call validate. And shall we make validate a static method for the CloudConfig?